### DISCUSSION TITLE
Bypass login screen on Hyper-V enhanced session

### DISCUSSION THREAD

**Discussion by tetsuya-ops** [neutrinolabs/xrdp#2432] [Q&A]
I just want to completely remove the blue screen and log-in automatically with xorg.
I have Ubuntu installed on hyper-v and when I start the enhanced session, it always displays the blue login screen. I don't want this to happen.
I already have saved my credentials on the xrdp.ini file, but it still displays the screen.
How can I remove it?
Thanks
![image](https://user-images.githubusercontent.com/62120571/202192959-e97216f7-7584-4ed0-a599-d616b86d44a3.png)

**Comment by matt335672** [comment]
You need to use an encrypted password in the ini file these days.

Have a look at this serverfault answer:-

https://serverfault.com/questions/867467/rdp-file-with-embedded-password-asks-for-password

**Reply by tetsuya-ops** [reply]
Thank you so much for the answer, but I still have the same problem.
The answer describes how to perform the encryption on windows.
But I need to do the same on an Ubuntu VM on the Hyper-V hypervisor.
Thank you so much.

**Reply by matt335672** [reply]
The encryption key is specific to your windows client (or account). Because of this you have to create the encrypted password string on the windows client you are using, and then put it in the RDP ini file.

When the client logs in, the password is decrypted and passed over the link to the xrdp server. The server will authenticate the user and skip the blue screen.

That's the mechanism anyway. Am I misunderstanding what you're trying to do?

**Reply by tetsuya-ops** [reply]
Understand... 
Is this the correct form that needs to be stored?
       [Xorg]
       name=Xorg
       lib=libxup.so
       username=myusername
       password=encrypted password
       ip=127.0.0.1
       port=-1
       code=20

**Reply by tetsuya-ops** [reply]
> Am I misunderstanding what you're trying to do?

No, you're right, that is exactly what I want to do, I just want to bypass the blue screen and login automatically. Thanks!!

**Reply by matt335672** [reply]
I can see the misunderstanding.

The username and encrypted password need to be stored on the Windows machine in the RDP ini file. I'm assuming you're using Windows clients.

On the Ubuntu side, just set username=ask and password-ask. They'll be ignored.

**Reply by tetsuya-ops** [reply]
No, you are not right there, I am using the enhanced session mode on Hyper-V. The RD client is integrated on the Hyper-V. So, I don't know if it is possible to change the ini file of Hyper-V enhanced session, if so, I don't know how.

**Reply by matt335672** [reply]
Ah, I see now. I'm afraid I'm not familiar with this facility.

Wouldn't having an auto-login like this working be rather insecure, in that anyone who connects to xrdp gets a free login to the VM?

**Reply by tetsuya-ops** [reply]
Sorry for being too late. I am the only one who connects to Ubuntu throw this way. I just want to have the same seamless login experience as I have when I use Windows over Hyper-V.

**Reply by matt335672** [reply]
Suggest you use an encrypted password as suggested above. That's pretty secure compared to other options.

**Reply by tetsuya-ops** [reply]
I know that is pretty much secure, but: Is there a way to avoid using the password and login screen?

**Reply by matt335672** [reply]
You have to authenticate for the system to work properly. For example, you may need keyrings to be unlocked.

**Comment by xgdgsc** [comment]
I also have to type password every time I login when using a windows client. Have you figured out why?
